homeassistant/components/recorder/db_schema.py

```diff
@@ -279,29 +307,47 @@ class States(Base):  # type: ignore[misc,valid-type]
         {"mysql_default_charset": "utf8mb4", "mysql_collate": "utf8mb4_unicode_ci"},
     )
     __tablename__ = TABLE_STATES
-    state_id = Column(Integer, Identity(), primary_key=True)
-    entity_id = Column(String(MAX_LENGTH_STATE_ENTITY_ID))
-    state = Column(String(MAX_LENGTH_STATE_STATE))
-    attributes = Column(
+    state_id: Mapped[int] = mapped_column(Integer, Identity(), primary_key=True)
+    entity_id: Mapped[str] = mapped_column(
+        String(MAX_LENGTH_STATE_ENTITY_ID), nullable=False
```

Comment:
Maybe we should leave this as nullable in case we ever get around to mapping the entity id to another table to avoid many copies of the string in the db

Reply:
Probably the same for event_type


Only matters if we would do it before SQLite requirements are new enough to have alter